### ISSUE TITLE
Blank Matter List

### ISSUE THREAD

**Issue by rdentandt** [Microsoft/mattercenter#841]
Hi All

After a tedious process of loading matter center I managed to successfully load it. I can create matters but I have the following issues:
1. No notifications are sent on a matter creation.
2. If I click on the matter direct link that is provided after a matter is create I can see the matter and create all calendar entries etc etc but If I browse matters on https://*****.azurewebsites.net/#/matters then there are no matters show. The list of previously created matters are blank.

Any suggestions on the above?

**Comment by jgm19** [comment]
I am experiencing the same issues. it just happened recently without any change done to the app. All of the previously created matters are not showing. even newly created ones. SPO search is working fine.

**Comment by jgm19** [comment]
@premchandpl can you help us out with this issue please?

**Comment by Vasanth1981** [comment]
Hi, We're also facing the same issue with Matter Center recently. Search Result is not listing newly created matters or documents. Re-Index has done multiple times including AppCatalog site too. But the Matter Creation is working fine, only issue with Search Result, (FindMatter or FindDocuments API calls)

The same issue happening in both our development or production environment.

-Vasanth.

**Comment by Vasanth1981** [comment]
@jgm19 Are you able to resolve this issue?

**Comment by Vasanth1981** [comment]
@WilsonReddy , Do you have any idea why the search matter would stop working after a certain time since its failed both development & Production at the same time. I'm using Matter Center 1st version.

**Comment by jgm19** [comment]
Hi @Vasanth1981 
No, I haven't resolved this issue yet. as of now, there seem to be no fix identified. I am wondering if this was caused by any changed in Sharepoint Online since the code for matter center web hasn't have an change for quite some time now and this functionality suddenly just stopped.

Anybody who can help? we cannot use matter center without this functionality because much of the feature can only be accessed through the search results.

**Comment by pjherman** [comment]
We are having the same issue.  I am able to create matters without any issues but they do not show up in the "All Matters" or "My Matters" list.  I have tried re-indexing and that has not helped.  I just happened to re-deploy a new list of clients and thought it was related to this process but now that I am seeing this thread, I believe the problem is bigger than this.  SharePoint search is working fine it is just the matter search that does not seem to be working.  Any help on this issue is greatly appreciated.

**Comment by Vasanth1981** [comment]
@pjherman Which version of Matter Center you're using? I just want to make sure it's not happening for Version 1.

**Comment by pjherman** [comment]
Version 2

**Comment by Vasanth1981** [comment]
Yes. I'm using Version 1. Looks like its affected for all the version of Matter Center. Just want to confirm it. Did you receive any support from Microsoft/Office 365 team?

**Comment by araver** [comment]
It's an issue with how Microsoft is crawling document libraries[https://github.com/SharePoint/sp-dev-docs/issues/2367](https://github.com/SharePoint/sp-dev-docs/issues/2367). I've created a pull request with an updated fix. [https://github.com/Microsoft/mattercenter/pull/843](https://github.com/Microsoft/mattercenter/pull/843)

**Comment by Vasanth1981** [comment]
@araver , For me (using Version 1), getting the document result is not an issue but Matters are not rending in the Result. Will this fix resolve both?

FindMatter method using below Constant,
 /// <summary>
 /// Constant required for document library filter condition
 /// </summary>
 public const string DocumentLibraryFilterCondition = "contentclass:STS_List_DocumentLibrary";

and FindDocument method using below Constant,
 /// <summary>
 /// Constant required for document item filter condition
 /// </summary>
 public const string DocumentItemFilterCondition = "contentclass:STS_ListItem_DocumentLibrary";

**Comment by araver** [comment]
@Vasanth1981 It should fix both.

**Comment by jgm19** [comment]
Hi @araver ! Thanks for this. Do i just over write the serviceconstants.cs file on the source and redeploy the webapp?

**Comment by jgm19** [comment]
@araver Sorry I'm new to github. I don;t use version control. I just downloaded the zip file and install it from there. If i just grab your version of the file and overwrite all copies on the source, how do i deploy that in the already deployed matter center in our test tenant? Should i reinstall mattercenter from zero?

thanks for all the help

**Comment by Vasanth1981** [comment]
@araver Sorry, No Luck. I'm using Version 1 and I tried following changes on the ConstantStrings.cs class under Utility project,

        /// <summary>
        /// Constant required for document library filter condition
        /// </summary>
        // public const string DocumentLibraryFilterCondition = "contentclass:STS_List_DocumentLibrary";
        // public const string DocumentLibraryFilterCondition = "or(contentclass:STS_List_DocumentLibrary,contentclass:STS_View)";
        // public const string DocumentLibraryFilterCondition = "or(contentclass:STS_ListItem_DocumentLibrary,contentclass:STS_View)";
        public const string DocumentLibraryFilterCondition = "contentclass:STS_View";

But none of them are working. Also, re-indexed the site.

-Vasanth.

**Comment by jgm19** [comment]
@Vasanth1981 did you have to reinstall mattercenter?

**Comment by Vasanth1981** [comment]
@jgm19  Nope. Just update the dll of Utility Project in Service class!

**Comment by araver** [comment]
@Vasanth1981 I'm not sure what the difference between version 1 and version 2. My fix is only applicable to the latest version of the source code.

**Comment by jgm19** [comment]
@Vasanth1981 sorry new here. So i overwrite the serviceconstants.cs file on the Microsoft.Legal.MatterCenter.Utility project, recompile and upload  the dll on the website's wwwroot? is it that simple. BTW i saw another instance of tghe serviceconstants.cs file in the \tree\master\cloud\src\solution\Microsoft.Legal.MatterCenter.jobs\Constants folder. do we need to update that too?

**Comment by Vasanth1981** [comment]
@jgm19 I'm not sure about it since I'm using Matter Center V1 its applicable to only one place.

**Comment by Vasanth1981** [comment]
@araver But the filter options for both version is same!

**Comment by araver** [comment]
@Vasanth1981 That may be true. But the code may be completely different. The variable names appear to be different.

**Comment by jgm19** [comment]
@Vasanth1981 , yes, this worked and fixed the issue for me. thanks @araver

**Comment by Vasanth1981** [comment]
@jgm19 , @araver 
The code fix is resolved the issue.
